tests/test_index_reader.py

```diff
@@ -396,6 +398,49 @@ def test_jstring_encoding(self):
         # the discussion here: https://github.com/castorini/pyserini/issues/770
         JString('zoölogy')
 
+    def test_dump_documents_BM25(self):
+        filepath = 'collections/cacm_documents_bm25_dump.json'
+        self.index_reader.dump_documents_BM25(filepath)
+        f = open(filepath, 'r')
+        dump = json.load(f)
+        f.close()
+
+        assert(len(dump) == self.index_reader.stats()['documents'])
+
+        query = 'Where can I learn to fly a helicopter?'
+        # Search through documents BM25 dump with query
+        query_terms = self.index_reader.analyze(query, analyzer=analysis.get_lucene_analyzer())
+        heap = [] # heapq implements a min-heap, we can invert the values to have a max-heap
+        for doc in dump:
+            score = 0
+            for term in query_terms:
+                if term in doc['vector']:
+                    score += doc['vector'][term]
+            heapq.heappush(heap, (-1*score, doc['id'])) 
+        # Using LuceneSearcher instead
+        hits = self.searcher.search(query)
+        for i in range(0, 10):
+            top = heapq.heappop(heap)
+            self.assertEqual(hits[i].docid, top[1])
+            self.assertAlmostEqual(hits[i].score, -1*top[0],places=3)
```

Comment:
pedantic nit, but space before 'places'.

Set up your code editor to de-lint for PEP8 compliance? should catch these.